src/Utility/Utils.cs

```diff
@@ -52,7 +59,75 @@ internal static PowerShell NewPwshInstance()
                 }
             }
 
-            return PowerShell.Create(s_iss);
+            var pwsh = PowerShell.Create(s_iss);
+            if (s_globalVariables == null)
+            {
+                // Get the names of the built-in global variables
+                ICollection<PSVariable> globalVars = GetGlobalVariables(pwsh);
+                s_globalVariables = new HashSet<string>(globalVars.Count + 3, StringComparer.OrdinalIgnoreCase)
```

Comment:
Could you please add a comment on what 3 means here?

Reply:
The comment is actually available right below this line :)

```
// These 3 variables are not in the built-in variables in a fresh Runspace,
// but they show up after we evaluate the 'profile.ps1' in the global scope.
```